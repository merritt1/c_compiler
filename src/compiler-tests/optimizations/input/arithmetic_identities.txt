void syscall_print_int(int integer);
void syscall_print_string(char *string);
int syscall_read_int(void);
void syscall_read_string(char *buffer, int length);
void syscall_exit(void);

void main(void) {
	int a;
	a = 123;
	a = a + 0;
	a = a - 0;
	a = a * 1;
	a = a / 1;
	a = a << 0;
	a = a >> 0;
	syscall_print_int(a);
}
