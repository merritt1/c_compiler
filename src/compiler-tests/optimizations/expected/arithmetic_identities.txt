(procBegin,  main)
(addressOf,     r0,      a)
(constInt,     r1,        123)
(storeWord,     r0,      r1)
(addressOf,     r2,      a)
(loadWord,     r3,      r2)
(parameter,         0,      r3)
(call    ,  _Global_syscall_print_int)
(procEnd ,  main)
